frontend/src/views/CellCards/components/CellCard/components/OntologyDagView/components/Node/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { MouseEventHandler } from "react";
+import { HierarchyPointNode } from "@visx/hierarchy/lib/types";
+import { Group } from "@visx/group";
+import { useRouter } from "next/router";
+import { ROUTES } from "src/common/constants/routes";
+import { TreeNodeWithState } from "../../common/types";
+import Text from "./components/Text";
+import RectOrCircle from "./components/RectOrCircle";
+
+export const CELL_CARD_ONTOLOGY_DAG_VIEW_CLICKABLE_TEXT_LABEL =
+  "cell-card-ontology-dag-view-clickable-text-label";
+
+type HierarchyNode = HierarchyPointNode<TreeNodeWithState>;
+
+interface NodeProps {
+  node: HierarchyNode;
+  handleClick?: MouseEventHandler<SVGGElement>;
+  isTargetNode: boolean;
+  handleMouseOver: (
+    event: React.MouseEvent<SVGElement>,
+    datum: TreeNodeWithState
+  ) => void;
+  handleMouseOut: () => void;
+  left: number;
+  top: number;
+  opacity: number;
+  animationKey: string;
+  maxWidth: number;
+  isInCorpus: boolean;
+}
+
+export default function Node({
+  node,
+  handleClick,
+  isTargetNode,
+  handleMouseOver,
+  handleMouseOut,
+  left,
+  top,
+  animationKey,
+  opacity,
+  maxWidth,
+  isInCorpus,
+}: NodeProps) {
+  const router = useRouter();
+
+  return (
+    <Group top={top} left={left} key={animationKey} opacity={opacity}>
+      {isInCorpus ? (
+        <g
+          data-testid={`${CELL_CARD_ONTOLOGY_DAG_VIEW_CLICKABLE_TEXT_LABEL}-${node.data.id}`}
+          onClick={() => {
```

Comment:
clicking on text label navigates to that cell card, but only if the label is in our corpus (`isInCorpus`)